### DISCUSSION TITLE
Option to raise an exception when a cassette needs to be re-recorded

### DISCUSSION THREAD

**Discussion by adeeb1** [vcr/vcr#859] [Ideas]
My tests run within a docker container in a Jenkins environment, and the test environment is reconstructed with each test run. As I understand it, if I first recorded a cassette a month ago and set the `re_record_interval` to 1 month, it will make a live API call every single time the cassette is run. This is because the cassette has expired, and although it updates the cassette file on the test environment, that update will not be committed to my repository. So every time the test environment is spun up again, it will see an expired cassette and make a live API call.

What do you think about adding an option to raise an exception if a cassette has expired and needs to be re-recorded? That would cause my test suite to fail, and a developer would have to manually re-record the cassette locally and commit it to the repository.

**Comment by krainboltgreene** [comment]
You can store cassettes elsewhere, see persistence documentation.
